pkg/discovery/grpcresolver/grpc_resolver.go

```diff
@@ -128,24 +124,39 @@ func (r *Resolver) rendezvousHash(addresses []string) []string {
 	for _, address := range addresses {
 		hosts = append(hosts, hostScore{
 			address: address,
-			score:   hashAddr(r.hasher, []byte(address), r.salt),
+			score:   hashAddr([]byte(address), r.salt),
 		})
 	}
 	sort.Sort(hosts)
-	addressesPerHost := make([]string, r.discoveryMinPeers)
-	for i := 0; i < r.discoveryMinPeers; i++ {
+	size := min(r.discoveryMinPeers, len(hosts))
+	addressesPerHost := make([]string, size)
+	for i := 0; i < size; i++ {
 		addressesPerHost[i] = hosts[i].address
 	}
 	return addressesPerHost
 }
 
-func hashAddr(hasher hash.Hash32, node, saltKey []byte) uint32 {
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func hashAddr(node, saltKey []byte) uint32 {
+	hasher := fnv.New32()
```

Comment:
I was suggesting to create it inside rendezvousHash(), so that it's reused for multiple addresses, but not kept across discovery updates